### PR TITLE
Add ability to hide modal from logged in customers. (Ticket #338939)

### PR DESCRIPTION
**Pull request by pratiknikam**
**Backend:**

image

### CODE REVIEW

File: app/code/community/SomethingDigital/Modal/etc/system.xml [hunk 1]
```diff
@@ -118,6 +118,16 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                         </skip_if_already_subscribed>
+                        <skip_if_already_logged_in translate="label commment">
+                            <label>Hide From Logged In Customers</label>
+                            <comment>Hide modal from customers who are already logged in.</comment>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <frontend_type>select</frontend_type>
+                            <sort_order>80</sort_order>
```
Comment (toddbc) [inline app/code/community/SomethingDigital/Modal/etc/system.xml]: nit: 100?

File: app/code/community/SomethingDigital/Modal/etc/config.xml [hunk 2]
```diff
@@ -36,6 +36,7 @@
                 <skip_cookie_check>0</skip_cookie_check>
                 <error_message>There was an error submitting the form. Please try again.</error_message>
                 <skip_if_already_subscribed>1</skip_if_already_subscribed>
+                <skip_if_already_logged_in>1</skip_if_already_logged_in>
```
Comment (mpchadwick) [inline app/code/community/SomethingDigital/Modal/etc/config.xml]: This should not default to on. It breaks BC (currently implementation is off)

Comment (pratiknikam) [inline app/code/community/SomethingDigital/Modal/etc/config.xml]: Ah ok. What is BC?

Comment (pratiknikam) [inline app/code/community/SomethingDigital/Modal/etc/config.xml]: Should we default to `Off `or remove it from default?

Comment (toddbc) [inline app/code/community/SomethingDigital/Modal/etc/config.xml]: Let's just remove the default.

BC = backwards compatibility.  Maybe someone already using SomethingDigital_Modal likes how it applies to logged in users.  Changing the default will affect them too, maybe surprisingly.

**Pull request merged**